onrobot_rg2_driver/src/rg2_driver_node.py

```diff
@@ -0,0 +1,123 @@
+            rospy.loginfo_throttle(5.0, 'Waiting for URscript Modbus watchdog activation...')
+            rate.sleep()
+
+        self._joint_publisher = rospy.Publisher("/joint_states", JointState, queue_size=1)
+
+        self._set_gripp_state_service = rospy.Service(
+            "rg2/set_gripper_width", GripperState, self._set_grip_state
+        )
+
+        rospy.Timer(rospy.Duration(0.05), self._pub_joint_state_cb)
+
+        rospy.loginfo(f'[{rospy.get_name()}] Node started')
+
+    def _set_grip_state(self, req: GripperStateRequest) -> GripperStateResponse:
+        try:
+            with self._lock:
+                if not self.MIN_RG2_FORCE <= req.force <= self.MAX_RG2_FORCE:
+                    raise ValueError(
+                        f'Force value must be between {self.MIN_RG2_FORCE} and {self.MAX_RG2_FORCE} [N]'
+                    )
+                if not self.MIN_RG2_RANGE <= req.width <= self.MAX_RG2_RANGE:
+                    raise ValueError(
+                        f'Width value must be between {self.MIN_RG2_RANGE} and {self.MAX_RG2_RANGE} [mm]'
+                    )
+
+                rospy.loginfo(
+                    f'[{rospy.get_name()}] Sending gripp request:\n'
+                    f'\t - width: {req.width}\n\t - force: {req.force}'
+                )
+
+                self._rg2_modbus.send_gripp_req(req.width, req.force)
+                grip_detected = self._wait_for_grip_completion(self.TIMEOUT)
+
+                rospy.loginfo(f'[{rospy.get_name()}] Grip action succes!')
+
+                return GripperStateResponse(True, grip_detected, None)
+
+        except (ValueError, TimeoutError, ModbusError, ConnectionError) as e:
+            rospy.logwarn(f'[{rospy.get_name()}] Grip action failed: {str(e)}')
+            return GripperStateResponse(False, False, str(e))
+
+    def _wait_for_grip_completion(self, timeout: float) -> bool:
+        start_time = rospy.Time.now()
+        rate = rospy.Rate(20)
+
+        while not self._rg2_modbus.querry_rg_busy_flag():
+            if (rospy.Time.now() - start_time).secs > timeout / 10:
+                # The preset position of the gripper is the same as 
+                # the actual position before the service was sent
+                return self._rg2_modbus.querry_grip_detected_flag()
+            rate.sleep()
+
+        rospy.loginfo(f'[{rospy.get_name()}] Request send succesful.')
+
+        while self._rg2_modbus.querry_rg_busy_flag():
+            if (rospy.Time.now() - start_time).secs > timeout:
+                raise TimeoutError('Grip action failed: timeout')
+            rate.sleep()
+
+        return self._rg2_modbus.querry_grip_detected_flag()
+
+    def _pub_joint_state_cb(self, *args) -> None:
+        try:
+            self._joint_state_msg.header.stamp = rospy.Time.now()
+
+            rg_width = self._rg2_modbus.querry_gripper_width()
+            joint_state = self._map(rg_width, 0.0, 101.0, -0.44, 1.0)
```

Comment:
```suggestion
            joint_state = self._map(rg_width, 0.0, 100.0, -0.44, 1.0)
```
Shouldn't be in range <0,100>?

Reply:
The gripper has some offset and this results in better accuracy :D